### ISSUE TITLE
Notifications not in order on MacOS

### ISSUE THREAD

**Issue by 5frank** [hbldh/bleak#155]
* bleak: 0.6.0a6 - **develop branch** 
* os: MacOS 10.14.6
* python: 3.7.5

### Description
logging received notification data in PeripheralDelegate.py directly after notifications enabled:
```
DEBUG:bleak.backends.corebluetooth.PeripheralDelegate:255: Read characteristic value b'\x18\x08\xb4\x98\x190\xe2\xa3\x03B\x07\xab(\xf46\x8c\x82\x02J\x05'
DEBUG:bleak.backends.corebluetooth.PeripheralDelegate:255: Read characteristic value b'\xf8\x01\xaa\x01M'
DEBUG:bleak.backends.corebluetooth.PeripheralDelegate:255: Read characteristic value b"\x18\x08\xb5\x98\x190\xb8\xa5\x03B\x07\xeb'\xfc6\xe8\x81\x02J\x05"
```
but the "user notifications callback" is not called in the same order.
```
b"\x18\x08\xb6\x98\x190\xe2\xa3\x03B\x07\xcf'\x807\xb6\x81\x02J\x05"
b'\x84\x02\xae\x01U'
```
- it only seems to occur directly after enabling notifications.
- I't is not due to some unexpected stdout linebuffering or similar.

I do not see how any recent change in the develop branch can have introduced this but I find it strange I have not seen this problem before.

**Comment by 5frank** [comment]
The issue is not always reproducible but my theory is that `peripheral_didUpdateValueForCharacteristic_error_` (that runs my callback)
runs while `asyncio.sleep(0.01)` is waiting for `peripheral_didUpdateNotificationStateForCharacteristic_error_` to set `_characteristic_notify_log[cUUID] = True`

```python
    async def startNotify_cb_(
        self, characteristic: CBCharacteristic, callback: Callable[[str, Any], Any]
    ) -> bool:
    [...]

        while not self._characteristic_notify_log[cUUID]:
            await asyncio.sleep(0.01)
```

this as removing `_characteristic_notify_log` and just check if there is a callback registered seems to fix it.

**Comment by 5frank** [comment]
Turns out this was rather hard to reproduce. It was reproducible for a moment when I had a PacketLogger running for days and AppStore consumed unreasonable amount of memory and CPU. I have not been able to recreate those conditions. 

I incorrectly stated packets was out of order - I believe it is only the first packet after enabling notifications that is lost. Also, I think this issue is not due to any recent change, I have indeed seen this problem before but thought it was related to some other issue.
